hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/sps/BlockStorageMovementNeeded.java

```diff
@@ -248,13 +250,18 @@ public void run() {
                   pendingWorkForDirectory.get(startINode);
               if (dirPendingWorkInfo != null
                   && dirPendingWorkInfo.isDirWorkDone()) {
-                ctxt.removeSPSHint(startINode);
                 pendingWorkForDirectory.remove(startINode);
+                ctxt.removeSPSHint(startINode);
```

Comment:
How about cathing and ignoring the exception here?
```suggestion
                try {
                  ctxt.removeSPSHint(startINode);
                } catch (FileNotFoundException e) {
                  // ignore if the file doesn't already exist
                  startINode = null;
                }
```

Reply:
@liubingxing I still prefer to catch the FileNotFoundException here. What do you think?

Reply:
@tasanuma Thanks for your suggestion. 
It is also a good way to solve this problem by catching the FileNotFoundException here.
I updated the code according to your suggestion.

Reply:
@liubingxing Thank you for updating the PR. I think we can keep the retry logic implemented before while catching the FileNotFoundException here.

Reply:
> @liubingxing I still prefer to catch the FileNotFoundException here. What do you think?

Sorry @tasanuma, I misunderstand your meaning here. I updated the code, please take a look.